### PR TITLE
Add Sentry to simplify debugging

### PR DESCRIPTION
**Pull request by Inkvi**
Will be very useful to debug errors in staging and prod environments.

## Summary by CodeRabbit

- **New Features**
  - Introduced a `GlobalError` component for enhanced error handling using Sentry.
  - Added Sentry integration for improved error tracking and monitoring across client, server, and edge environments.

- **Chores**
  - Updated `.gitignore` to include `.env.sentry-build-plugin`.
  - Added `@sentry/nextjs` package to dependencies.

- **Configuration**
  - Updated `next.config.mjs` for Sentry integration with various configuration options.
  - Configured Sentry initialization for client, server, and edge environments.

### CODE REVIEW

File: sentry.client.config.ts
Comment (mvpoyatt) [inline sentry.client.config.ts:1]: Does this file get used anywhere?

**Comment by Inkvi** [comment]
Not directly, AFAIK it’s just a Sentry config.

**Review by mvpoyatt — APPROVED** [review]

File: sentry.client.config.ts
[the remark this answers is not in the document]
Comment (Inkvi) [reply sentry.client.config.ts:1]: That file is autogenerated by sentry and is used on the client side

File: instrumentation.ts
[the remark this answers is not in the document]
Comment (Inkvi) [reply instrumentation.ts:1]: That file is autogenerated by sentry and I don't want to mess with it

**Pull request merged**